### PR TITLE
Unblock dotnet/sdk build with small fixes

### PR DESCRIPTION
**Pull request by nguerrera**
1. dotnet was not added to PATH so building with full msbuild would not use a dotnet that was downloaded based on global.json
2. The condition for default minimum VS version was inverted, it would ignore the version in global.json and use 15.9 always.
3. Make it possible to locate a given version of VS for other purposes without requiring it and using it for all builds

### CODE REVIEW

**Comment by nguerrera** [comment]
Found an issue, will reopen when fixed

**Comment by jcagme** [comment]
I haven’t looked at all the chances since I’m in my phone but noticed you are not working on eng/common/tools.sh. We try to kept both in sync.

**Comment by nguerrera** [comment]
These are all Windows specific changes. 1 is so that VS msbuild can find sdk. 2 and 3 relate to finding VS

**Comment by nguerrera** [comment]
I've now made (1) x-plat for consistency 

(2) and (3) have no equivalent at all x-plat, they are entirely about locating VS.

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -136,6 +136,11 @@ function InitializeDotNetCli([bool]$install) {
     $env:DOTNET_INSTALL_DIR = $dotnetRoot
   }
 
+  # Add dotnet to PATH. This prevents any bare invocation of dotnet in custom
+  # build steps from using anything other than what we've downloaded.
+  # It also  ensures that VS msbuild will use the downloaded sdk targets.
```
Comment (tmat) [inline eng/common/tools.ps1]: nit: extra space

File: eng/common/tools.ps1 [hunk 2]
```diff
@@ -169,12 +174,13 @@ function InstallDotNetSdk([string] $dotnetRoot, [string] $version) {
 # Returns full path to msbuild.exe.
 # Throws on failure.
 #
-function InitializeVisualStudioMSBuild([bool]$install) {
+function InitializeVisualStudioMSBuild([bool]$install, [object]$vs = $null) {
```
Comment (tmat) [inline eng/common/tools.ps1]: I'd call the new parameter `vsRequirements`, so that it's clearer what it means.

File: eng/common/tools.ps1 (hunk 1 above)
Comment (nguerrera) [inline eng/common/tools.ps1]:
```suggestion
  # It also ensures that VS msbuild will use the downloaded sdk targets.
```

File: eng/common/tools.ps1 [hunk 3]
```diff
@@ -255,7 +261,7 @@ function InstallXCopyMSBuild([string] $packageVersion) {
 # Returns JSON describing the located VS instance (same format as returned by vswhere), 
 # or $null if no instance meeting the requirements is found on the machine.
 #
-function LocateVisualStudio {
+function LocateVisualStudio([object]$vs = $null){
```
Comment (tmat) [inline eng/common/tools.ps1]: The same here

**Pull request merged**